src/main/java/org/radarcns/phone/PhoneUsageManager.java

```diff
@@ -0,0 +1,289 @@
+        phoneStateFilter.addAction(Intent.ACTION_SCREEN_OFF); // lock
+        phoneStateFilter.addAction(Intent.ACTION_SHUTDOWN); // shutdown
+        getService().registerReceiver(new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                // If previous event was a shutdown, then this action indicates that the phone has booted
+                if (preferences.getString(LAST_USER_INTERACTION,"").equals(Intent.ACTION_SHUTDOWN)) {
+                    sendInteractionState(ACTION_BOOT, System.currentTimeMillis() / 1000d);
+                }
+
+                sendInteractionState(intent.getAction(), System.currentTimeMillis() / 1000d);
+            }
+        }, phoneStateFilter);
+
+        updateStatus(DeviceStatusListener.Status.CONNECTED);
+    }
+
+
+    public void sendInteractionState(String action, double timestamp) {
+        PhoneInteractionState state;
+
+        switch (action) {
+            case Intent.ACTION_SCREEN_OFF:
+                state = PhoneInteractionState.STANDBY;
+                break;
+            case Intent.ACTION_USER_PRESENT:
+                state = PhoneInteractionState.UNLOCKED;
+                break;
+            case Intent.ACTION_SHUTDOWN:
+                state = PhoneInteractionState.SHUTDOWN;
+                break;
+            case ACTION_BOOT:
+                state = PhoneInteractionState.BOOTED;
+                break;
+            default:
+                return;
+        }
+
+        double timestampReceived = System.currentTimeMillis() / 1000d;
+        PhoneUserInteraction value = new PhoneUserInteraction(
+                timestamp, timestampReceived, state);
+        send(userInteractionTable, value);
+
+        // Save the last user interaction state. Value shutdown is used to register boot.
+        preferences.edit()
+                .putString(LAST_USER_INTERACTION, action)
+                .apply();
+        logger.info("Interaction State: {} {}", timestamp, state);
+    }
+
+    public final synchronized void setUsageEventUpdateRate(final long period) {
+        // Create an intent and alarm that will be wrapped in PendingIntent
+        Intent intent = new Intent("ACTIVITY_LAUNCH_WAKE");
+
+        // Create the pending intent and wrap our intent
+        PendingIntent pendingIntent = PendingIntent.getBroadcast(this.context, 1, intent, 0);
+
+        // Get alarm manager and schedule it to run every period (seconds)
+        AlarmManager alarmManager = (AlarmManager) this.context.getSystemService(ALARM_SERVICE);
+        alarmManager.setInexactRepeating(AlarmManager.RTC_WAKEUP, System.currentTimeMillis(), period * 1000, pendingIntent);
+
+        // Activity to perform when alarm is triggered
+        this.context.registerReceiver(new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                processUsageEvents();
```

Comment:
Have you tested this call while the app is not running? I'm wondering if the service will get started to actually process the message, or that you'd get errors that the current context is not valid.